src/backend/utils/adt/ri_triggers.c

```diff
@@ -1597,11 +1596,11 @@ RI_Initial_Check(Trigger *trigger, Relation fk_rel, Relation pk_rel)
 	 * register the snapshot, because SPI_execute_snapshot will see to it. We
 	 * need at most one tuple returned, so pass limit = 1.
 	 */
-	spi_result = SPI_execute_snapshot(qplan,
+	RUN_AS_PSQL(spi_result = SPI_execute_snapshot(qplan,
```

Comment:
I think this is no better than original one. Let us keep original one only. I thought we want do it at one place in caller but that does not seems like the case unless we do something like this
func()
{
    SET_PSQL_DIALECT();
    function body
    RESET_PSQL_DIALECT();
}

#define SET_PSQL_DIALECT 
 set dialect;
PG_TRY();
 {
        

#define RESET_PSQL_DIALECT
}
PG_CATCH()
{
     reset dialect;
}

which is again is not ideal.